libazureinit/src/provision/ssh.rs

```diff
@@ -43,123 +45,225 @@ pub(crate) fn provision_ssh(
     Ok(())
 }
 
-pub(crate) fn get_authorized_keys_path_from_sshd() -> io::Result<Option<PathBuf>>
-{
-    let sshd_output = Command::new("sshd").arg("-T").output()?;
-    if !sshd_output.status.success() {
-        return Err(io::Error::new(
-            io::ErrorKind::Other,
-            format!("sshd -T failed with status: {}", sshd_output.status),
-        ));
+fn get_sshd_g_output() -> io::Result<Output> {
+    Command::new("sshd").arg("-G").output()
+}
+
+fn get_authorized_keys_path_from_sshd(
+    sshd_config_command_runner: impl Fn() -> io::Result<Output>,
+) -> Option<String> {
+    let output: Output = match run_sshd_command(sshd_config_command_runner) {
+        Some(output) => output,
+        None => {
+            info!("Unable to determine authorizedkeysfile from sshd -G");
+            return None;
+        }
+    };
+
+    match extract_authorized_keys_file_path(&output.stdout) {
+        Some(path) => Some(path),
+        None => {
+            info!("No authorizedkeysfile in sshd configuration");
+            None
+        }
     }
+}
 
-    let stdout = sshd_output.stdout;
-    let sshd_output = String::from_utf8_lossy(&stdout);
-    for line in sshd_output.lines() {
+fn run_sshd_command(
+    sshd_config_command_runner: impl Fn() -> io::Result<Output>,
+) -> Option<Output> {
+    match sshd_config_command_runner() {
+        Ok(output) if output.status.success() => {
+            info!(
+                "sshd -G succeeded with stdout length={:?}",
+                output.stdout.len()
+            );
+            Some(output)
+        }
+        Ok(output) => {
+            error!(
+                "sshd -G failed with code={} stdout={:?} stderr={:?}",
+                output.status,
+                String::from_utf8_lossy(&output.stdout),
+                String::from_utf8_lossy(&output.stderr)
+            );
```

Comment:
FYI tracing events and spans can have [structured fields.](https://docs.rs/tracing/latest/tracing/#recording-fields) You might want to use at least the output status as a field:

```Rust
            error!(
                ?output.status,
                "sshd -G failed with stdout={:?} stderr={:?}",
                String::from_utf8_lossy(&output.stdout),
                String::from_utf8_lossy(&output.stderr)
            );
```

This should also get translated into an OpenTelemetry field (or whatever their verbiage is) - then you should be able to write queries against it easily

Reply:
Awesome! took that lesson and applied it across PR